### PR TITLE
Update test_cmd_line from Cpython v3.11.2

### PR DESCRIPTION
**Pull request by Masorubka1**

### CODE REVIEW

**Comment by DimitrisJim** [comment]
A couple of successes:

```bash
UNEXPECTED SUCCESS: test_optimize (test.test_cmd_line.CmdLineTest.test_optimize)
UNEXPECTED SUCCESS: test_output_newline (test.test_cmd_line.CmdLineTest.test_output_newline)
UNEXPECTED SUCCESS: test_site_flag (test.test_cmd_line.CmdLineTest.test_site_flag)
```

and a couple of failures:

```bash
FAIL: test_tokenizer_error_with_stdin (test.test_cmd_line.SyntaxErrorTests.test_tokenizer_error_with_stdin)
FAIL: test_decoding_error_at_the_end_of_the_line (test.test_cmd_line.SyntaxErrorTests.test_decoding_error_at_the_end_of_the_line)
FAIL: test_xoption_frozen_modules (test.test_cmd_line.CmdLineTest.test_xoption_frozen_modules) []
FAIL: test_xoption_frozen_modules (test.test_cmd_line.CmdLineTest.test_xoption_frozen_modules) [=]
FAIL: test_xoption_frozen_modules (test.test_cmd_line.CmdLineTest.test_xoption_frozen_modules) [=on]
FAIL: test_int_max_str_digits (test.test_cmd_line.CmdLineTest.test_int_max_str_digits)
FAIL: test_help_xoptions (test.test_cmd_line.CmdLineTest.test_help_xoptions)
FAIL: test_help_env (test.test_cmd_line.CmdLineTest.test_help_env)
FAIL: test_help_all (test.test_cmd_line.CmdLineTest.test_help_all)
FAIL: test_help (test.test_cmd_line.CmdLineTest.test_help)
ERROR: test_non_interactive_output_buffering (test.test_cmd_line.CmdLineTest.test_non_interactive_output_buffering)
```

File: Lib/test/test_cmd_line.py [hunk 1]
```diff
@@ -459,6 +459,8 @@ def test_stdin_readline(self):
             "import sys; print(repr(sys.stdin.readline()))",
             b"'abc\\n'")
 
+    # TODO: RUSTPYTHON
+    @unittest.skipIf(sys.platform.startswith('win'), "TODO: RUSTPYTHON windows has \n troubles")
```
Comment (fanninpm) [inline Lib/test/test_cmd_line.py:463]: I think other tests are marked `TODO: RUSTPYTHON, universal newlines`, but I like this description better, as it's a clearer statement of the problem.

**Comment by youknowone** [comment]
2 more windows issues

```
FAIL: test_output_newline (test.test_cmd_line.CmdLineTest.test_output_newline)
FAIL: test_stdin_readline (test.test_cmd_line.CmdLineTest.test_stdin_readline)
```

**Review by DimitrisJim — APPROVED** [review]
thanks! lgtm!

**Pull request merged**